code/VideoGroup.php

```diff
@@ -37,138 +42,38 @@ class VideoGroup extends Page
-
-        // Build a list of all IDs for VideoGroups that are children
-        $holderIDs = $this->VideoGroupIDs();
-
-        if ($holderIDs) {
-            if ($filter) {
-                $filter .= ' OR ';
-            }
-            $filter .= '"ParentID" IN (' . implode(',', $holderIDs) . ")";
-        }
-
-        $order = '"SiteTree"."Title" ASC';
-
-        $entries = Video::get()->where($filter)->sort($order);
-
-        $list = new PaginatedList($entries, Controller::curr()->request);
-        $list->setPageLength($limit);
-        return $list;
+        $groupIDS = $this->VideoGroupIDS();
+        $groupIDS[] = $this->ID;
+        return Video::get()->filter('ParentID', $groupIDS)->sort(array('Title' => 'ASC'));
 
     }
 
 
     /**
-     * old display function
-     *
-     * @return DataList
+     * @return ArrayList
      */
-    public function ShowVideos()
+    public function ChildGroups()
     {
-
-        $filter = '';
-
-        $limit = (isset($_GET['start']) && (int)$_GET['start'] > 0) ? (int)$_GET['start'] . "," . self::$page_length : "0," . self::$page_length;
-        $sort = (isset($_GET['sortby'])) ? Convert::raw2sql($_GET['sortby']) : "\"Title\"";
-
-        $groupids = array($this->ID);
-
-        if (self::$include_child_groups && $childgroups = $this->ChildGroups(true)) {
-            $groupids = array_merge($groupids, $childgroups->map('ID', 'ID'));
-        }
-
-        $groupidsimpl = implode(',', $groupids);
-
-        $join = $this->getManyManyJoin('Videos', 'Video');
-        $multicatfilter = $this->getManyManyFilter('Videos', 'Video');
-
-        //TODO: get products that appear in child groups (make this optional)
-
-        $products = DataObject::get('Video', "(\"ParentID\" IN ($groupidsimpl) OR $multicatfilter) $filter", $sort,
-            $join, $limit);
-
-        $allproducts = DataObject::get('Video', "\"ParentID\" IN ($groupidsimpl) $filter", "", $join);
-
-        if ($allproducts) {
-            $products->TotalCount = $allproducts->Count();
-        } //add total count to returned data for 'showing x to y of z products'
-        if ($products && $products instanceof DataObjectSet) {
-            $products->removeDuplicates();
-        }
-
-
-        return $products;
-
-    }
-
-    /**
-     * @param bool|false $recursive
-     * @return DataList|mixed|null
-     */
-    public function ChildGroups($recursive = false)
-    {
-        if ($recursive) {
-            if ($children = DataObject::get('VideoGroup', "\"ParentID\" = '$this->ID'")) {
-                $output = unserialize(serialize($children));
-                foreach ($children as $group) {
-                    $output->merge($group->ChildGroups($recursive));
-                }
-                return $output;
-            }
-            return null;
-        } else {
-            return DataObject::get('VideoGroup', "\"ParentID\" = '$this->ID'");
-        }
+        return $this->Children()->filter('ClassName', 'VideoGroup');
```

Comment:
Should we be returning `AllChildren()` here? There may be an inconsistency as `VideoGroupIDs()` returns `AllChildren()`'s ID's.